File: crates/lsh/tests/fixtures/rust/kitchen_sink.rs
// Line comment
//! Inner doc
/// Outer doc
/* block comment */

use std::collections::HashMap;

const MAX: u32 = 42;

#[derive(Clone, Debug)]
pub struct Point<'a> {
    pub name: &'a str,
    pub value: f64,
}

fn main() {
    let s = "hello";
    let raw = r#"raw"#;
    let c = 'x';
    let n: i64 = 0xff_u32 as i64;
    let v = vec![1, 2, 3];

    if let Some(x) = v.first() {
        println!("{}", x);
    } else {
        return;
    }

    match n {
        0 => (),
        _ => unreachable!(),
    }
}

File: crates/lsh/tests/fixtures/go/kitchen_sink.go
// Line comment
/* block comment */

package main

import (
    "fmt"
    "strings"
)

const MAX = 42

type Point struct {
    Name  string
    Value float64
}

func main() {
    s := "hello"
    raw := `raw string`
    r := 'x'
    n := 0xff
    xs := []int{1, 2, 3}

    for i, v := range xs {
        if v == 0 {
            continue
        }
        fmt.Println(i, v, strings.ToUpper(s), raw, r, n)
    }

    switch n {
    case 0:
        return
    default:
        fmt.Println("default")
    }
}

func builtins() {
    s := make([]string, 0)
    s = append(s, "x")
    n := len(s)
    c := cap(s)
    _ = copy(s, s)
    delete(m, "key")
    if ok := true; ok {
        panic("boom")
    }
    _ = new(int)
    _ = complex(1, 2)
    _ = real(c)
    _ = imag(c)
    _ = close(ch)
    v := recover()
    _ = v
    println("print")
    print("no newline")
}

var (
    b bool
    by byte
    e error
    a any
    c comparable
)
